iotedgedev/cli.py

```diff
@@ -663,14 +670,18 @@ def list_subscriptions_and_set_default():
     if not azure_cli.list_subscriptions():
         sys.exit()
     default_subscriptionId = azure_cli.get_default_subscription()
-    return default_subscriptionId
+    output.info(f"Select an Azure Subscription Name or Id [{default_subscriptionId}]:")
+    return input() or default_subscriptionId
```

Comment:
>default_subscriptionId [](http://example.com/codeflow?start=22&length=22)

azure_cli.get_default_subscription(), no need for intermediate variable

Reply:
not sure, since `azure_cli.get_default_subscription()` actually calls `az account show`, I think it may be better to execute that only once. wdyt?

Reply:
if we want to execute az accoutn show once and preserve the azure_cli abstraction layer i think we should cache the response the first time when it is called. I am happy to leave it for a separate PR too.

Reply:
yes, feel free to create a tracking issue, thanks so much